test/container_manager/canary_utils.go

```diff
@@ -0,0 +1,109 @@
+package container_manager
+
+import (
+	"flag"
+	"github.com/Shopify/sarama"
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/strimzi/strimzi-canary/internal/config"
+	"github.com/strimzi/strimzi-canary/internal/security"
+	"github.com/strimzi/strimzi-canary/internal/servers"
+	"github.com/strimzi/strimzi-canary/internal/services"
+	"github.com/strimzi/strimzi-canary/internal/workers"
+	"log"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+)
+
+var (
+	clientCreationFailed = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name:      "client_creation_error_total",
+		Namespace: "strimzi_canary",
+		Help:      "Total number of errors while creating Sarama client",
+	}, nil)
+)
+
+// TODO goes against DRY principle, possible refactor of cmd/main to put more of code into none main module
+func createCanary() {
+	canaryConfig := config.NewCanaryConfig()
+
+	// Always log to stderr by default
+	if err := flag.Set("logtostderr", "true"); err != nil {
+		glog.Errorf("Error on setting logtostderr to true")
+	}
+	flag.Set("v", strconv.Itoa(canaryConfig.VerbosityLogLevel))
+	flag.Parse()
+
+	glog.Infof("Starting Strimzi canary tool with config: %+v", canaryConfig)
+
+	httpServer := servers.NewHttpServer()
+	httpServer.Start()
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+
+	client, err := newClient(canaryConfig)
+	if err != nil {
+		glog.Fatalf("Error creating new Sarama client: %v", err)
+	}
+
+	topicService := services.NewTopicService(canaryConfig, client)
+	producerService := services.NewProducerService(canaryConfig, client)
+	consumerService := services.NewConsumerService(canaryConfig, client)
+
+	canaryManager := workers.NewCanaryManager(canaryConfig, topicService, producerService, consumerService)
+	canaryManager.Start()
+
+	sig := <-signals
+	glog.Infof("Got signal: %v", sig)
+	canaryManager.Stop()
+	httpServer.Stop()
```

Comment:
Guess it might be useful to defer these two calls to `Stop()` right
after the servers have been started.